internal/namingschema/cache.go

```diff
@@ -0,0 +1,41 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023 Datadog, Inc.
+
+package namingschema
+
+import "fmt"
+
+// CacheSystem represents cache systems to be used for cache naming schemas in this package.
+type CacheSystem string
+
+const (
+	// CacheSystemMemcached represents Memcached.
+	CacheSystemMemcached CacheSystem = "memcached"
+)
+
+type cacheOutboundOperationNameSchema struct {
+	cfg    *config
+	system CacheSystem
+}
+
+// NewCacheOutboundOperationNameSchema creates a new naming schema for outbound operations from caching systems.
+func NewCacheOutboundOperationNameSchema(system CacheSystem, opts ...Option) *Schema {
+	cfg := &config{}
+	for _, opt := range opts {
+		opt(cfg)
+	}
+	return New(&cacheOutboundOperationNameSchema{cfg: cfg, system: system})
+}
+
+func (c *cacheOutboundOperationNameSchema) V0() string {
+	if f, ok := c.cfg.versionOverrides[SchemaV0]; ok {
+		return f()
+	}
+	return c.V1()
```

Comment:
I'm not sure I understand why the `V0` returns `V1` here! What's the value of implementing the `V0`? If it's for overriding, can't we just implement the `V1` and support overrides there?
Since `V0` is defined as the current version, I'd expect it to return the current naming, e.g `memcached.query` instead of `memcached.command`.
(applies to all implementations not only cache.go)

Reply:
The problem with `v0` is that mostly doesn't follow any structure/pattern among integrations (with exception of http clients/servers).
For that reason, the `V0` implementation of cache is just going to `v1`, because it will have to be overwritten in most of the integrations using this.

> Since V0 is defined as the current version, I'd expect it to return the current naming, e.g memcached.query instead of memcached.command.
The issue is that this is also gonna be used by other integrations classified as cache (like Redis) - so there isn't a behavior I can put here that adapts to all of them to fit for `v0`.

Please let me know if what I wrote above makes sense to you 😄

Reply:
Thanks for explaining! In this case I don't think we need a V0 implementation at all since there is no common structure/pattern. For simplicity I'd go with a V1 implementation that supports an override.

Reply:
Feel free to resolve as we discussed it offline. Let's just make sure we mention in the doc string of V0 that it returns V1 if the override wasn't set.

Reply:
Thanks! will clarify on the doc